spring-cloud-gcp-core/src/main/java/org/springframework/cloud/gcp/core/DefaultGcpProjectIdProvider.java

```diff
@@ -0,0 +1,40 @@
+/*
+ *  Copyright 2017 original author or authors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.core;
+
+import java.util.Optional;
+
+import com.google.cloud.ServiceOptions;
+
+/**
+ * A project ID provider that wraps {@link ServiceOptions#getDefaultProjectId()}.
+ *
+ * @author João André Martins
+ */
+public class DefaultGcpProjectIdProvider implements GcpProjectIdProvider {
```

Comment:
@artembilan @meltsufin how about this?

It feels weird to me to add a `String`-based ctor here since, if a user has a static project ID, they can just go `() -> "my-project-id"`.